### DISCUSSION TITLE
Google Cloud Database Migration Service for implicitly created tables

### DISCUSSION THREAD

**Discussion by armandoborge** [prisma/prisma#8771] [Q&A]
Hi there!

I need to migrate a Postgres database instance hosted on Google (Cloud SQL) from one region to another, but according the GCP docs:  Database Migration Service migrates only tables with primary keys. Any tables on the source PostgreSQL database without primary key constraints won't be migrated. For these tables, Database Migration Service will migrate only the table schema.

See here: https://cloud.google.com/database-migration/docs/postgres/configure-source-database?#configure-your-source-databases

My database schema has a implicitly created table (many to many), this table doesn't have a primary key but has relevant data for the application, so I need to include this data in the migration.  What would be the proper way to convert the implicit table to a explicit one and add a primary key without losing any data?

My scheme is like this:

```
model Institution {
  id               String         @id @db.VarChar(120)
  countries  Country[]   @relation("InstitutionCountry")
}

model Country {
  code              String          @id @db.Char(3)
  name             String           @db.VarChar(60)
  institutions   Institution[]  @relation("InstitutionCountry")
}
```

Appreciate any help!

**Comment by janpio** [comment]
Have you considered just adding the primary key temporarily and removing it again after the migration? That would leave your Prisma schema and application code unchanged.

**Reply by armandoborge** [reply]
Neat!
I didn't consider it before.
Thanks for your suggestion, this seems the way to go.

**Reply by janpio** [reply]
It certainly adds a small bit of risk during the time of the migration that the application will not liek that additional Primary Key, but as far as I am aware just running Prisma Client queries should not be impacted at all. Only if you `db pull` or run Migrations in any way you need to be careful of course.

**Reply by armandoborge** [reply]
To minimize risk I'll stop all the CRUD operations and migrations during the migration period.

**Reply by janpio** [reply]
If that is an option, that is indeed great.

**Reply by armandoborge** [reply]
On development and staging environments is for sure a option, I'll need to measure a production migration case to check if we can afford the downtime.

**Reply by janpio** [reply]
(Might also be worth to just test your application with the PK added, as I said as long as you only use Prisma Client with CRUD and similar queries, I _think_ this should work just fine)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
